nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java

```diff
@@ -0,0 +1,300 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.fixserialization;
+
+import com.google.common.base.Preconditions;
+import com.sun.source.util.Trees;
+import com.uber.nullaway.fixserialization.qual.AnnotationFactory;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import javax.lang.model.element.Element;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+/** Config class for Fix Serialization package. */
+public class FixSerializationConfig {
+
+  /**
+   * If activated, for all reporting errors, NullAway will serialize information and suggests type
+   * changes to resolve them, in case these errors could be fixed by adding a @Nullable annotation.
+   * These data will be written at output directory.
+   */
+  public final boolean suggestEnabled;
+  /**
+   * If activated, serialized information of a fix suggest will also include the enclosing method
+   * and class of the element involved in error.
+   */
+  public final boolean suggestEnclosing;
+  /** If activated, NullAway will write reporting errors in output directory. */
+  public final boolean logErrorEnabled;
+  /** If activated, errors information will also include the enclosing method and class. */
+  public final boolean logErrorEnclosing;
+  /** The directory where all files generated/read by Fix Serialization package resides. */
+  public final String outputDirectory;
+
+  public final AnnotationFactory annotationFactory;
+  /**
+   * Contains a set of classes fully qualified name. If not empty, NullAway will only report errors
+   * in these classes.
+   */
+  public final Set<String> workList;
+
+  public final Writer writer;
```

Comment:
`Writer` is already a well-known name in the Java standard library.  We need a more specific class name here.

Reply:
Renamed to ```Serializer```. Please let me know if that doesn't work. db12b596f807699fe73d310bd6aba7bb6de9ce6d